xonsh/tools.py

```diff
@@ -317,18 +317,42 @@ class redirect_stderr(_RedirectStream):
     _stream = "stderr"
 
 
+def _executables_in_posix(path):
+    if PYTHON_VERSION_INFO < (3, 5, 0):
+        for fname in os.listdir(path):
+            fpath  = os.path.join(path, fname)
+            if (os.path.exists(fpath) and os.access(fpath, os.X_OK) and \
+                                    (not os.path.isdir(fpath))):
+                yield fname
+    else:
+        yield from (x.name for x in scandir(path)
+                    if x.is_file() and os.access(x.path, os.X_OK))
```

Comment:
Here, `x` mihgt not exist. **and** `scandir` is 3.5+ so I'm not sure the change is desirable. (while I have nothing against 3.5 only, I'm not sure about the project in general.